src/models/room_user.model.js

```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+let RoomUserSchema = new Schema({
+  roomId: { type: Schema.Types.ObjectId, required: true, ref: "Room" },
+  userId: { type: Schema.Types.ObjectId, required: true, ref: "User" }
+});
+
+RoomUserSchema.index({ roomId: 1, userId: 1 }, { unique: true }); //compound unique index
+
+RoomUserSchema.statics.findByRoomId = async function(roomId) {
```

Comment:
you can just return the promise itself and not use an async function so like this:
```
findByRoomId = function(roomId){
  return this.find({ roomId })
}
```
```

Reply:
aye aye